app/src/main/java/com/example/films/MainActivity.kt

```diff
@@ -1,21 +1,42 @@
 package com.example.films
 
-import android.content.Intent
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(),
+    FragmentMoviesList.TransactionsFragmentMLClicks,
+    FragmentMoviesDetails.TransactionsFragmentMDClicks {
 
     override fun onCreate(savedInstanceState: Bundle?) {
+
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        val button : Button = findViewById(R.id.main_activity_button)
+        this.supportActionBar?.hide()
 
-        button.setOnClickListener {
-            val intent = Intent(this@MainActivity, MovieDetailsActivity::class.java)
-            startActivity(intent)
+        if (savedInstanceState == null) {
+
+            supportFragmentManager.beginTransaction()
+                .add(R.id.fragment_container, FragmentMoviesList())
+                .commit()
         }
     }
+
+    override fun filmOnClick() {
+
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.fragment_container, FragmentMoviesDetails())
+            .addToBackStack(null)
+            .commit()
+    }
+
+    override fun backOnClick() {
+
+        supportFragmentManager.popBackStack()
+
+        supportFragmentManager.beginTransaction()
```

Comment:
to back movie list not need create instance movieList movieList fragment
just `supportFragmentManager.popBackStack()` enough

Reply:
done